src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PatientMapperTest.java

```diff
@@ -1,19 +1,207 @@
+
+    @Mock
+    private InvestigationSubject investigationSubject;
+
+    @Mock
+    private PatientDetails patientDetails;
+
+    @Test
+    void testMapMessageToPatientThrowsExceptionWhenServiceReportDetailsAreNull() {
+        assertThatThrownBy(() -> patientMapper.mapToPatient(message))
+            .isInstanceOf(FhirValidationException.class)
+            .hasMessage("Unable to map message to patient details");
+    }
+
+    @Test
+    void testMapMessageToPatientThrowsExceptionWhenInvestigationSubjectIsNull() {
+        when(message.getServiceReportDetails()).thenReturn(serviceReportDetails);
+
+        assertThatThrownBy(() -> patientMapper.mapToPatient(message))
+            .isInstanceOf(FhirValidationException.class)
+            .hasMessage("Unable to map message to patient details");
+    }
+
+    @Test
+    void testMapMessageToPatientThrowsExceptionWhenPatientDetailsAreNull() {
+        when(message.getServiceReportDetails()).thenReturn(serviceReportDetails);
+        when(serviceReportDetails.getSubject()).thenReturn(investigationSubject);
+
+        assertThatThrownBy(() -> patientMapper.mapToPatient(message))
+            .isInstanceOf(FhirValidationException.class)
+            .hasMessage("Unable to map message to patient details");
+    }
 
     @Test
     void testMapMessageToPatient() {
-        final Message message = new Message(new ArrayList<>());
+        when(message.getServiceReportDetails()).thenReturn(serviceReportDetails);
+        when(serviceReportDetails.getSubject()).thenReturn(investigationSubject);
+        when(investigationSubject.getDetails()).thenReturn(patientDetails);
+
+        stubPersonName();
+        stubGender();
+        stubDateOfBirth(BIRTH_DATE_YYMMDD, DateFormat.CCYYMMDD);
+
+        final Patient patient = patientMapper.mapToPatient(message);
+
+        assertAll("patient",
 
-        assertThat(new PatientMapper().mapToPatient(message)).isExactlyInstanceOf(Patient.class);
+            () -> assertThat(patient).isNotNull(),
+
+            () -> assertThat(patient.getIdentifier()).hasSize(1).first()
+                .satisfies(identifier -> assertAll(
+                    () -> assertThat(identifier.getValue()).isEqualTo(NHS_NUMBER),
+                    () -> assertThat(identifier.getSystem()).isEqualTo(PatientMapper.NHS_NUMBER_SYSTEM))),
+
+            () -> assertThat(patient.getName()).hasSize(1).first()
+                .satisfies(name -> assertAll(
+                    () -> assertThat(name.getPrefixAsSingleString()).isEqualTo(TITLE),
+                    () -> assertThat(name.getGivenAsSingleString()).isEqualTo(FIRST_FORENAME + " " + SECOND_FORENAME),
+                    () -> assertThat(name.getFamily()).isEqualTo(SURNAME))),
+
+            () -> assertThat(patient.getGender()).isEqualTo(Enumerations.AdministrativeGender.FEMALE),
+
+            () -> {
+                final Date birthDate = Date.from(LocalDate.of(1999, 12, 28).atStartOfDay()
```

Comment:
My usual points, that this test does not make it clear why these values should be expected without the reader having to leave the method. Do `all` patients get the same `NHS_NUMBER`?

Can we at least amend the `stubXXX()` methods to accept the values you'll be verifying later?

Similarly, it's not immediately obvious why `BIRTH_DATE_YYMMDD` and `CCYYMMDD` should map specifically to `1999`, `12`, `28`, but a little thought (without leaving the method!) lets us make a reasonable assumption quickly.

Reply:
I've refactored the test cases, hopefully they make more sense. Let me know if you have any further suggestions